pkg/ddevapp/traefik.go

```diff
@@ -394,19 +402,113 @@ func configureTraefikForApp(app *DdevApp) error {
 
-		err = t.Execute(f, templateData)
+		err = t.Execute(dynamicConfigTemp, templateData)
 		if err != nil {
 			return fmt.Errorf("could not parse traefik_config_template.yaml with templatedate='%v':: %v", templateData, err)
 		}
+
+		tmpFileName := dynamicConfigTemp.Name()
+		err = dynamicConfigTemp.Close()
+		if err != nil {
+			return err
+		}
+
+		/* The following section reads the project/.ddev/traefik/dynamic_config.*.yaml files, fills any template placeholders in them with the
+		App's templateData (for	targeting the appropriate routers (e.g. {projectname}-web-80-http) or for rewriting the App's URL in the
+		response body), then merges their content into the base dynamic config YAML generated above, and is finally written to /project/.ddev/
+		traefik/config/<project>.yaml. Allows for adding middlewares, overriding settings, etc...
+		*/
+
+		// find all dynamic_config.*.yaml files in the project's .ddev/traefik directory
+		extraDynamicConfigFiles, err := fileutil.GlobFilenames(projectTraefikDir, "dynamic_config.*.yaml")
+		if err != nil {
+			return err
+		}
+		var finalYaml []byte
+		// Check if extra config files were found. If not, no need to proceed.
+		if len(extraDynamicConfigFiles) > 0 {
+
+			// convert config files to maps and merge them, returning a yaml string
+			resultYaml, err := util.MergeYamlFiles(tmpFileName, extraDynamicConfigFiles...)
+			if err != nil {
+				return err
+			}
+
+			// In the event that any of the extra configs contained go template {{ }} placeholders, create a new template and parse the YAML string into it.
+			tmpl, err := template.New("dynamic_config_extras").Funcs(getTemplateFuncMap()).Parse(string(resultYaml))
+			if err != nil {
+				return fmt.Errorf("error parsing template: %s", err)
+			}
+
+			// Execute the template with the app's templateData
+			var extraConfigProcessedYAML strings.Builder
+			err = tmpl.Execute(&extraConfigProcessedYAML, templateData)
+			if err != nil {
+				return fmt.Errorf("error executing template: %s", err)
+			}
+
+			// convert the output to a string and prepend "#ddev-generated" to the string, and then convert all of it to []byte to be written to file
+			finalYaml = []byte("#ddev-generated\n" +
+				"# If you remove the ddev-generated line above you\n" +
+				"# are responsible for maintaining this file. DDEV will not then\n" +
+				"# update it, for example if you add `additional_hostnames`, etc.\n" +
+				"# However, it is now unnecessary and imprudent to take control of\n" +
+				"# this file, as there is now a dynamic merge mechanism. To use it,\n" +
+				"# create <project>/.ddev/traefik/dynamic_config.*.yaml files which\n" +
+				"# will then be merged in alphanumeric order. See the readme in the\n" +
+				"# /traefik directory for more information\n\n" +
+				extraConfigProcessedYAML.String())
+
+		} else {
+
+			// Since there weren't any dynamic_config.*.yaml files, we can just write the contents of the base temp file to .ddev/traefik/<project.yaml>
```

Comment:
This gets much simpler as suggested above, always process the finalYaml, no "else" required.